pkg/prometheus/operator.go

```diff
@@ -457,56 +456,69 @@ func ListOptions(name string) api.ListOptions {
 //
 // TODO(fabxc): remove this once the PetSet controller learns how to do rolling updates.
 func (c *Operator) syncVersion(key string, p *spec.Prometheus) error {
-	podClient := c.kclient.Core().Pods(p.Namespace)
-
-	pods, err := podClient.List(ListOptions(p.Name))
+	status, oldPods, allReady, err := PrometheusStatus(c.kclient, p)
 	if err != nil {
 		return err
 	}
 
 	// If the PetSet is still busy scaling, don't interfere by killing pods.
 	// We enqueue ourselves again to until the PetSet is ready.
-	if len(pods.Items) != int(p.Spec.Replicas) {
+	if status.Replicas != p.Spec.Replicas {
 		return fmt.Errorf("scaling in progress")
 	}
-	if len(pods.Items) == 0 {
+	if status.Replicas == 0 {
+		return nil
+	}
+	if len(oldPods) == 0 {
 		return nil
 	}
+	if !allReady {
```

Comment:
That check (and bool return argument) can probably be replaced by checking `status.UnavailableReplicas > 0`.